sherpa/optmethods/optfcts.py

```diff
@@ -273,31 +224,29 @@ def difevo_lm(fcn, x0, xmin, xmax, ftol=EPSILON, maxfev=None, verbose=0,
     x, xmin, xmax = _check_args(x0, xmin, xmax)
 
     # make sure that the cross over prob is within [0.1,1.0]
-    xprob = max(0.1, xprob)
-    xprob = min(xprob, 1.0)
+    xprob = float(np.clip(xprob, 0.1, 1.0))
 
     # make sure that weighting_factor is within [0.1,1.0]
-    weighting_factor = max(0.1, weighting_factor)
-    weighting_factor = min(weighting_factor, 1.0)
+    weighting_factor = float(np.clip(weighting_factor, 0.1, 1.0))
 
     if population_size is None:
         population_size = 16 * x.size
 
     if maxfev is None:
         maxfev = 1024 * x.size
 
+    # TODO: can we not just call x.size rather than
+    #       np.asanyarray(fcn(x)).size for the last argument?
+    #
```

Comment:
If `fcn` does not change the size of the array, the answer is clearly : yes.

Reply:
Unfortunately I'm not convinced we can guarantee we have that condition here [it would be easy to check but even then there's enough going on with this code, including the fact we currently don't seem to have any tests of it, that I'm reluctant to change things too much.

Reply:
This is not a bit time sink, so not a problem.